Makefile

```
@@ -243,6 +243,32 @@ endif
 
 pgbackrest-pgimg-docker: pgbackrest-pgimg-build
 
+# ----- Special case image (upgrade) -----
+
+# Special case args: UPGRADE_PG_VERSIONS (defines all versions of PG that will be installed) 
+upgrade-img-build: ccbase-image $(CCPROOT)/build/upgrade/Dockerfile
+	$(IMGCMDSTEM) \
+		-f $(CCPROOT)/build/upgrade/Dockerfile \
+		-t $(CCP_IMAGE_PREFIX)/crunchy-upgrade:$(CCP_IMAGE_TAG) \
+		--build-arg BASEOS=$(CCP_BASEOS) \
+		--build-arg BASEVER=$(CCP_VERSION) \
+		--build-arg PG_FULL=$(CCP_PG_FULLVERSION) \
+		--build-arg PG_MAJOR=$(CCP_PGVERSION) \
+		--build-arg PREFIX=$(CCP_IMAGE_PREFIX) \
+		--build-arg DFSET=$(DFSET) \
+		--build-arg PACKAGER=$(PACKAGER) \
+		--build-arg UPGRADE_PG_VERSIONS="$(shell find $(CCPROOT)/conf -type f -name "crunchypg*.repo" | \
+			grep -o [1-9][0-9] | tr '\n' ' ')" \
```

⛏️ I think `$(shell)` already replaces newlines with space. https://www.gnu.org/software/make/manual/html_node/Shell-Function.html#Shell-Function

```suggestion
			grep -o [1-9][0-9])" \
```

---

Yup, that works.  Updated.